### PR TITLE
implement setuptools-scm

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  other
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-4511
- *Research reference*:  na

### Verification and Testing

pip installed the repo and got the fancy version back
image

### CODE REVIEW

**Review by patricktnast — APPROVED** [review]

File: src/vivarium_nih_us_cvd/_version.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__version__ = "0.1.dev353+gb79e496.d20231016"
```
Comment (mattkappel) [inline src/vivarium_nih_us_cvd/_version.py]: We don't want to commit this file

File: src/vivarium_nih_us_cvd/components/risk_correlation.py [hunk 2]
```diff
@@ -68,7 +68,9 @@ def get_population_attributable_fraction_source(
             "risk_factor.joint_mediated_risks.population_attributable_fraction"
         )
         pafs = {}
-        for (name, measure), group in paf_data.groupby(["affected_entity", "affected_measure"]):
+        for (name, measure), group in paf_data.groupby(
```
Comment (mattkappel) [inline src/vivarium_nih_us_cvd/components/risk_correlation.py:71]: Is this unrelated?

Comment (mattkappel) [reply src/vivarium_nih_us_cvd/components/risk_correlation.py:71]: oh, it's a `black` reformat

**Review by mattkappel — APPROVED** [review]
Looks good except you don't want to commit _version.py.

File: src/vivarium_nih_us_cvd/_version.py (hunk 1 above)
Comment (stevebachmeier) [reply src/vivarium_nih_us_cvd/_version.py]: @mattkappel  I was wondering about that. HOw does this all work w/o committing it?

Comment (stevebachmeier) [reply src/vivarium_nih_us_cvd/_version.py]: And do we add this to the .gitignore then?

**Pull request merged**